### PR TITLE
Beachfront: Fix Shared Memory Overwriting

### PR DESCRIPTION
**Pull request by guscarreon**
The Beachfront adapter seems to be writting to shared memory inside the `getVideoRequests` function, particulartly to the `bfReqs[i].Request.Device` and the `bfReqs[i].Request.Site` fields. This pull request proposes changes to stop the race conditions, but authors of the beachfront adapter source code should be notified and encouraged to make any changes to this PR if they disagree with the proposed approach.

### CODE REVIEW

File: adapters/beachfront/beachfront.go [hunk 1]
```diff
@@ -437,10 +443,11 @@ func getVideoRequests(request *openrtb.BidRequest) ([]beachfrontVideoRequest, []
 			}
 		}
 
-		if bfReqs[i].Request.Device != nil && bfReqs[i].Request.Device.DeviceType == 0 {
+		if deviceCopy != nil && deviceCopy.DeviceType == 0 {
```
Comment (SyntaxNode) [inline adapters/beachfront/beachfront.go:446]: It's not possible for `deviceCopy` to be `nil`.

Comment (guscarreon) [reply adapters/beachfront/beachfront.go:446]: Yep, I noticed, left the check anyways but I can remove

**Comment by guscarreon** [comment]
@muncha,
Recently coded pull request #1756 is meant to enable Prebid Server to find data race conditions in adapter bidder source code. According to this newly coded feature, your adapter might be writing to shared memory and this PR implements an approach to prevent it. Please let us know whether or not you agree with the changes proposed in this pull request or if you rather want to code the fix yourself. 

When it comes to adapters' source code improvements, we usually wait a week for owners to come back to us. If we get no feedback, we would go ahead and review and merge this pull request ourselves.

We'd love to hear your take on this issue.

File: adapters/beachfront/beachfront.go (hunk 1 above)
Comment (SyntaxNode) [reply adapters/beachfront/beachfront.go:446]: Cool. Might as well clean that up while we're here. Likely left over from an older implementation.

**Comment by muncha** [comment]
@guscarreon - Thanks, this looks great. I've got nothing to add.

**Review by SyntaxNode — APPROVED** [review]

File: adapters/beachfront/beachfront.go [hunk 2]
```diff
@@ -407,22 +407,27 @@ func getVideoRequests(request *openrtb.BidRequest) ([]beachfrontVideoRequest, []
 		bfReqs[i].Request = *request
 		var secure int8
 
+		var deviceCopy openrtb.Device
 		if bfReqs[i].Request.Device == nil {
-			bfReqs[i].Request.Device = &openrtb.Device{}
+			deviceCopy = openrtb.Device{}
```
Comment (hhhjort) [inline adapters/beachfront/beachfront.go:412]: Doesn't line 410 already do this? I think you only need one of these two lines.

**Review by hhhjort — APPROVED** [review]
LGTM

**Pull request merged**